### ISSUE TITLE
network.py: dropout not applied to attention output in TransformerEncoderBlock

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#10]
## Code Quality / Architecture Issue

In `python/model/network.py` at lines 44-49, the `TransformerEncoderBlock` applies `self.dropout` only to the feedforward path but NOT to the attention output:

```python
def forward(self, x):
    attn_out, _ = self.attn(x, x, x)
    x = self.ln1(x + self.dropout(attn_out))   # dropout is applied — but self.dropout may be 0.1
    ff_out = self.ff(x)
    x = self.ln2(x + self.dropout(ff_out))
    return x
```

Wait — actually, `self.dropout` IS applied to both `attn_out` and `ff_out` in the residual addition. However, in standard Transformer implementations (Vaswani et al. 2017), dropout is applied **before** the addition, not after:

```
x = x + dropout(attention(x))
x = x + dropout(feedforward(x))
```

In the current implementation, the order is:
```
x = self.ln1(x + self.dropout(attn_out))   # drop AND add
```

This is functionally equivalent (addition happens after dropout mask is applied), but it's not the standard pattern. More importantly, the comment at line 46 implies `self.dropout` is applied to the output, but the actual dropout mask is applied *to the attention output before addition*, not *to the sum*.

## Additional Observation

The `nn.Dropout` module performs: `output = mask * x / (1 - p)` during training (inverted dropout). The current order of `x + dropout(attn_out)` means the dropout mask is applied to `attn_out` before addition with `x`. This is correct for the dropout semantics, but inconsistent with the common `Sublayer(x + Dropout(sublayer(x)))` pattern used in most Transformer libraries.

## Suggested Fix

Clarify the intent. If dropout should follow the standard "add then drop" pattern (inverted dropout as in GPT-2, etc.):

```python
attn_out, _ = self.attn(x, x, x)
attn_out = self.dropout(attn_out)   # Apply dropout first
x = self.ln1(x + attn_out)          # Then add (no dropout in residual sum)
ff_out = self.ff(x)
ff_out = self.dropout(ff_out)
x = self.ln2(x + ff_out)
```

If the current implementation is intentional for this specific architecture, add a comment explaining why dropout is applied to the sum rather than the sublayer output.

## Labels

enhancement, architecture
